### PR TITLE
pkg/ocicni: a cross-platform namespace manager thing

### PR DESCRIPTION
**Pull request by vbatts**
Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

**Comment by vbatts** [comment]
rebased. PTAL.

**Comment by vbatts** [comment]
@rajatchopra ping

File: pkg/ocicni/util_unsupported.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// +build !linux
```
Comment (rajatchopra) [inline pkg/ocicni/util_unsupported.go]: What are we going to do for Windows?

Comment (vbatts) [reply pkg/ocicni/util_unsupported.go]: Add support for it? For now it's a stub to get things compiling. Your experience would be most helpful 🤩

Comment (dcbw) [reply pkg/ocicni/util_unsupported.go]: @vbatts the grand plan is to remove the need for nsenter entirely and just rely on the CNI plugin and/or cached state.  There's really no reason to touch the network namespace from ocicni.

Comment (vbatts) [reply pkg/ocicni/util_unsupported.go]: @dcbw that sounds best. Is there a timeline on that?

**Pull request merged**